During ACKNOWLEDGE get 'java.lang.IllegalStateException: Content stream is already disposed.'

Time to time I got this exception trying to send ACKNOWLEDGE (not every time):
```java
Caused by: java.lang.IllegalStateException: Content stream is already disposed.
	at io.reactivex.netty.protocol.http.UnicastContentSubject$OnSubscribeAction.call(UnicastContentSubject.java:240)
	at io.reactivex.netty.protocol.http.UnicastContentSubject$OnSubscribeAction.call(UnicastContentSubject.java:213)
	at rx.Observable.unsafeSubscribe(Observable.java:9861)
	at rx.internal.operators.OnSubscribeMap.call(OnSubscribeMap.java:48)
	at rx.internal.operators.OnSubscribeMap.call(OnSubscribeMap.java:33)
	at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:48)
	at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:30)
	at rx.Observable.subscribe(Observable.java:9957)
	at rx.Observable.subscribe(Observable.java:9924)
	at rx.Observable.subscribe(Observable.java:9729)
	at rx.Observable.forEach(Observable.java:6781)
	at com.mesosphere.mesos.rx.java.SinkSubscriber.lambda$onNext$7(SinkSubscriber.java:66)
	at rx.internal.util.ActionSubscriber.onNext(ActionSubscriber.java:39)
	at rx.observers.SafeSubscriber.onNext(SafeSubscriber.java:134)
	at rx.internal.operators.OperatorObserveOn$ObserveOnSubscriber.call(OperatorObserveOn.java:227)
	... 9 more
```

--- a/mesos-rxjava-client/src/main/java/com/mesosphere/mesos/rx/java/SinkSubscriber.java
+++ b/mesos-rxjava-client/src/main/java/com/mesosphere/mesos/rx/java/SinkSubscriber.java
@@ -29,6 +29,7 @@ import rx.functions.Func1;
 
 import java.util.List;
 import java.util.Map;
+import java.util.Optional;
 
 final class SinkSubscriber<Send> extends Subscriber<SinkOperation<Send>> {
 
@@ -51,19 +52,17 @@ final class SinkSubscriber<Send> extends Subscriber<SinkOperation<Send>> {
             final Send toSink = op.getThingToSink();
             createPost.call(toSink)
                 .flatMap(httpClient::submit)
-                .observeOn(Rx.compute())
-                .subscribe(resp -> {
+                .flatMap(resp -> {
                     final HttpResponseStatus status = resp.getStatus();
                     final int code = status.code();
 
                     if (code == 202) {
                         /* This is success */
-                        op.onCompleted();
+                        return Observable.just(Optional.<MesosException>empty());
                     } else {
                         final HttpResponseHeaders headers = resp.getHeaders();
-                        ResponseUtils.attemptToReadErrorResponse(resp)
-                            .observeOn(Rx.compute())
-                            .forEach(msg -> {
+                        return ResponseUtils.attemptToReadErrorResponse(resp)
+                            .map(msg -> {
                                 final List<Map.Entry<String, String>> entries = headers.entries();
                                 final MesosClientErrorContext context = new MesosClientErrorContext(code, msg, entries);
                                 MesosException error;
@@ -77,8 +76,16 @@ final class SinkSubscriber<Send> extends Subscriber<SinkOperation<Send>> {
                                     // something else that isn't success but not an error as far as http is concerned
                                     error = new MesosException(toSink, context);
                                 }
-                                op.onError(error);
+                                return Optional.of(error);
                             });
+                    }
+                })
+                .observeOn(Rx.compute())
+                .subscribe(exception -> {
+                    if (!exception.isPresent()) {
+                        op.onCompleted();
+                    } else {
+                        op.onError(exception.get());
                     }
                 });
         } catch (Throwable e) {

